pom.xml

```diff
@@ -579,6 +553,92 @@ limitations under the License.]]></inlineHeader>
     </plugins>
   </build>
   <profiles>
+    <profile>
+      <id>jdk11</id>
+      <activation>
+        <jdk>[9..)</jdk>
+      </activation>
+      <build>
+        <pluginManagement>
+          <plugins>
+            <plugin>
+              <artifactId>maven-compiler-plugin</artifactId>
+              <version>3.8.1</version>
+              <configuration>
+                <source>${java.version}</source>
+                <target>${java.version}</target>
+                <release>${java.release.version}</release>
+                <optimize>true</optimize>
+                <showDeprecation>true</showDeprecation>
+                <showWarnings>true</showWarnings>
+                <failOnWarning>false</failOnWarning>
+                <fork>false</fork>
+                <compilerArgs>
+                  <arg>-XDcompilePolicy=simple</arg>
+                  <arg>-Xplugin:ErrorProne</arg>
+                </compilerArgs>
+                <annotationProcessorPaths>
+                  <path>
+                    <groupId>com.google.errorprone</groupId>
+                    <artifactId>error_prone_core</artifactId>
+                    <version>2.4.0</version>
+                  </path>
+                </annotationProcessorPaths>
+              </configuration>
+            </plugin>
+          </plugins>
+        </pluginManagement>
+      </build>
+    </profile>
+    <profile>
+      <id>jdk8</id>
+      <activation>
+        <jdk>1.8</jdk>
+      </activation>
+      <properties>
+        <javac.version>9+181-r4173-1</javac.version>
```

Comment:
why for the profile `1.8` you are using version `9.xx`?

Reply:
this is the version of the ErrorProne compiler, it does not affect how Maven activate profiles.
this is the version suggested by ErrorProne docs